client/Components/Button.tsx

```diff
@@ -21,20 +21,22 @@ export function Button(props: ButtonProps) {
 
   const classNames = ["c-button"];
 
-  if (props.disabled) {
-    classNames.push("c-button--disabled");
-  }
+  const faElement = props.fontAwesomeIcon && (
```

Comment:
Interesting place to put this! I think this is a bit too much of a hack. This will take a bit of architecture to get right. The Command class api would need to take an optional render method for the FontAwesomeIcon, which would let you put this logic in BuildEncounterCommandList where it probably belongs. I could get this done if you can think of one or two more use cases for having a variable icon.

Reply:
You'll get no argument from me. This felt clunky, for sure, but I didn't want to rely purely on CSS, just in case you ever swap out icons or something.

As for variable icons, I could certainly see use cases, but few so contextually simple as a toggle.

* when the library pane is open, swap the `fa-book-medical` with a `fa-book-open` (I know that's currently used for the library manager, but I could easily see that being `fa-database` or one of the many list patterns
* it's silly, but I'd love to add `fa-spin` to the `fa-cog` while the settings modal is open, just so it's spinning quietly in the background as a visual cue of what's open
* we don't currently have an "unlink initiative" (though I've wanted one on more than one occasion), but the `fa-unlink` is an obvious ringer there.
* I'd love to have the "next turn" button's icon, when it starts back at the top of initiative, to swap out with `fa-redo fa-spin` for a few seconds, just to reflect it's starting back at the top.

I can imagine a few others offhand, but they're less specific to the icons themselves (like visually disabling move up/down at the ends of the lists, or disabling clean encounter when it's already blank, etc.).

Clearly, though, none of those are quite as simple as toggling the menu..

Reply:
Ok! This is really helpful. I'll think about this, I think there's a way to get a nice API for this kind of stuff.